src/wallet.rs

```diff
@@ -237,6 +244,65 @@ pub fn send_xch(
     Ok(ctx.take())
 }
 
+pub fn create_dig_collateral_coin(
+    collateral_dig_coins: Vec<Cat>,
+    morphed_store_id: Bytes32,
+    synthetic_key: PublicKey,
+    fee_coins: Vec<Coin>,
+    fee: u64,
+) -> Result<Vec<CoinSpend>, WalletError> {
+    let mut collateral_amount = 0_u64;
+    for cat in &collateral_dig_coins {
+        if cat.info.asset_id != DIG_COIN_ASSET_ID {
+            return Err(WalletError::Driver(DriverError::InvalidAssetId));
+        }
+        collateral_amount += cat.coin.amount;
+    }
+
+    let p2_parent_hash = P2ParentCoin::inner_puzzle_hash(Some(DIG_COIN_ASSET_ID));
```

Comment:
to avoid confusion, I usually include 'inner' for inner puzzle hashes - e.g., 'p2_parent_coin_inner_puzzle_hash' (calling this p2 parent coin is my fault)